ffmpeg/ffmpeg_test.go

```diff
@@ -374,3 +376,81 @@ func TestTranscoder_Timestamp(t *testing.T) {
 	`
 	run(cmd)
 }
+
+func TestTranscoder_Statistics(t *testing.T) {
+	// Checks the stats returned after transcoding
+
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+
+	cmd := `
+        set -eux
+        cd $0
+
+        # prepare 1-second input
+        cp "$1/../transcoder/test.ts" inp.ts
+        ffmpeg -loglevel warning -i inp.ts -c:a copy -c:v copy -t 1 test.ts
+    `
+	run(cmd)
+
+	// set a 60fps input at a small resolution (to help runtime)
+	p144p60fps := P144p30fps16x9
+	p144p60fps.Framerate = 60
+	// odd / nonstandard input just to sanity check.
+	podd123fps := VideoProfile{Resolution: "124x70", Framerate: 123, Bitrate: "100k"}
+
+	// Construct output parameters.
+	// Quickcheck style tests would be nice here one day?
+	profiles := []VideoProfile{P240p30fps16x9, P144p30fps16x9, p144p60fps, podd123fps}
+	out := make([]TranscodeOptions, len(profiles))
+	for i, p := range profiles {
+		out[i] = TranscodeOptions{Profile: p, Oname: fmt.Sprintf("%s/out%d.mp4", dir, i)}
+	}
+
+	res, err := Transcode3(&TranscodeOptionsIn{Fname: dir + "/test.ts"}, out)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for i, r := range res {
+		w, h, err := VideoProfileResolution(out[i].Profile)
+		if err != nil {
+			t.Error(err)
+		}
+
+		// Check pixel counts
+		if r.Pixels != int64(w*h*r.Frames) {
+			t.Error("Mismatched pixel counts")
+		}
+
+		// Check frame counts against ffprobe-reported output
+
+		// First, generate stats file
+		f, err := os.Create(fmt.Sprintf("%s/out%d.res.stats", dir, i))
+		if err != nil {
+			t.Error(err)
+		}
+		b := bufio.NewWriter(f)
+		fmt.Fprintf(b, `[STREAM]
```

Comment:
style nit: please newline before the backtick if golang allows it.  (it might not, in which case, ignore this comment)

Reply:
After go fmt, the indentation is still strange:

```diff
[nix-shell:~/livepeer/code/github.com/livepeer/lpms/ffmpeg]$ git diff
diff --git a/ffmpeg/ffmpeg_test.go b/ffmpeg/ffmpeg_test.go
index 868eb20..a8756d2 100644
--- a/ffmpeg/ffmpeg_test.go
+++ b/ffmpeg/ffmpeg_test.go
@@ -435,7 +435,8 @@ func TestTranscoder_Statistics(t *testing.T) {
                        t.Error(err)
                }
                b := bufio.NewWriter(f)
-               fmt.Fprintf(b, `[STREAM]
+               fmt.Fprintf(b,
+                       `[STREAM]
 width=%d
 height=%d
 nb_read_frames=%d
```

I'm inclined to keep it as-is but can change if you prefer.

(The whole reason for the weird formatting here is because we test on `diff` output and golang doesn't have heredoc-style indentation adjustment. Originally I ignored whitespace via `diff -Wbu` but that felt somewhat fragile, so went this route.)

Reply:
I think its better to go with go fmt, but I agree with you, it's weird either way.

VS code might try to autoformat it.  We should probably just commit whatever VSCode allows.  Do you use VSCode?

Reply:
vim for life :) `gofmt` is run after every save, so this is already "gofmt clean".

Does VSCode use a different formatter?

Reply:
No, VSCode uses same formatter. On re-saving this file it doesn't change formatting.